dlt/sources/helpers/rest_client/paginators.py

```diff
@@ -100,84 +267,95 @@ class OffsetPaginator(BasePaginator):
         client = RESTClient(
             base_url="https://api.example.com",
             paginator=OffsetPaginator(
-                initial_limit=100,
+                limit=100,
                 total_path="total"
             )
         )
         @dlt.resource
         def get_items():
-            for page in client.paginate("/items", params={"limit": 100}):
+            for page in client.paginate("/items"):
                 yield page
 
-    Note that we pass the `limit` parameter in the initial request to the API.
     The `OffsetPaginator` will automatically increment the offset for each
     subsequent request until all items are fetched.
+
+    If the API does not provide the total count of items, you can use the
+    `maximum_offset` parameter to limit the number of items to fetch. For example:
+
+        client = RESTClient(
+            base_url="https://api.example.com",
+            paginator=OffsetPaginator(
+                limit=100,
+                maximum_offset=1000
+            )
+        )
+        ...
+
+    In this case, pagination will stop after fetching 1000 items.
     """
 
     def __init__(
         self,
-        initial_limit: int,
-        initial_offset: int = 0,
+        limit: int,
+        offset: int = 0,
         offset_param: str = "offset",
         limit_param: str = "limit",
         total_path: jsonpath.TJsonPath = "total",
+        maximum_offset: Optional[int] = None,
+        initial_limit: Optional[int] = None,
+        initial_offset: Optional[int] = None,
     ) -> None:
         """
         Args:
-            initial_limit (int): The maximum number of items to retrieve
+            limit (int): The maximum number of items to retrieve
                 in each request.
-            initial_offset (int): The offset for the first request.
+            offset (int): The offset for the first request.
                 Defaults to 0.
             offset_param (str): The query parameter name for the offset.
                 Defaults to 'offset'.
             limit_param (str): The query parameter name for the limit.
                 Defaults to 'limit'.
             total_path (jsonpath.TJsonPath): The JSONPath expression for
                 the total number of items.
+            maximum_offset (int): The maximum offset value. If provided,
+                pagination will stop once this offset is reached or exceeded,
+                even if more data is available. This allows you to limit the
+                maximum range for pagination. Defaults to None.
+            initial_limit (int, optional): Deprecated, use `limit` instead.
+            initial_offset (int, optional): Deprecated, use `offset` instead.
         """
-        super().__init__()
-        self.offset_param = offset_param
-        self.limit_param = limit_param
-        self.total_path = jsonpath.compile_path(total_path)
+        if initial_limit is not None:
+            warnings.warn(
```

Comment:
we have really good system for deprecations, look for `DltDeprecationWarning` not only does this but also shows usage of such functions in mypy and VSCode :)